modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -1,17 +1,17 @@
 package edu.gemini.seqexec.engine
 
+import edu.gemini.seqexec.model.SharedModel.SequenceMetadata
+
 import scalaz._
 import Scalaz._
 
 /**
   * A list of `Step`s grouped by target and instrument.
   */
-case class Sequence[+A](id: String, steps: List[Step[A]])
+case class Sequence[+A](id: String, metadata: SequenceMetadata, steps: List[Step[A]])
```

Comment:
Right, I also toyed with the idea of making metadata just a `B` type, but that seems to cause trouble on the functor definition